TESTING.adoc

```diff
@@ -0,0 +1,91 @@
+[[addons-tests]]
+= Add-ons Tests
+:icons:
+:toc: macro
+:toc-title:
+:toclevels: 1
+
+toc::[]
+
+[[about]]
+== Introduction
+This repository uses integration test suite from link:https://github.com/minishift/minishift[`minishift/minishift`] repository to test the functionality of the add-ons.
+The test suite is written in Go and uses Gherkin language for definition of test sets.
+This attitude allows add-on contributors to write tests effectively using predefined understandable sentences, for example:
+
+```gherkin
+  Scenario: Applying the CORS add-on
+    Given Minishift has state "Running"
+     When executing "minishift addons apply cors" succeeds
+     Then stdout should contain "CORS is now allowed from any address"
+```
+
+NOTE: We recommend to align to the ends of Gherkin keywords `Given`, `When`, `Then`, `And`.
+
+For more information about Gherkin you can check its link:https://github.com/cucumber/cucumber/wiki/Gherkin[wiki page].
+More in-depth information about Minishift integration test suite can be found at respective part of link:https://docs.openshift.org/latest/minishift/contributing/developing.html#integration-tests[Minishift documentation].
+
+=== Structure of tests
+Feature files which define the test cases for each individual addon are stored in `test/integration/features` directory.
+The feature files follow naming convention of `<addon-name>.feature` and begins with the tag `@<addon-name>`.
+
+Integration tests are implemented in `test/integration/integration_test.go` file which imports test suite from Minishift repository and can be used for import of additional step definitions.
+
+To help with execution of integration tests `make integration` target is provided in the Makefile.
+Its usage is covered later in this document. 
+
+== Prerequisites
+To run the test successfully you need to:
+
+- install link:https://docs.openshift.org/latest/minishift/getting-started/installing.html[Minishift and its prerequisites]
+- install recent link:https://golang.org/dl/[Go distribution] (>=1.8)
+- install link:https://github.com/golang/dep[dep package], run: `go get -u github.com/golang/dep/cmd/dep`
+- git clone this repository into `$GOPATH/src/github.com/minishift/minishift-addons`
+- get latest minishift binary from https://github.com/minishift/minishift/releases/latest and put it into `testing/bin`
+- vendor needed Go packages by runining: `make vendor` in root directory of the `minishift-addons` repository
+
+NOTE: We recommend to vendor the packages once a while to make sure you use up to date packages.
```

Comment:
How often? "Once in a while" can mean many different spans of time to different people.